wordpress-mobile/WordPress-Android
# Handling  deleting an uploading media in Aztec

Fixes #6456 

This PR takes care of 2 situations:
- when a user deletes one or more media items within Aztec, [now Aztec triggers an event](https://github.com/wordpress-mobile/AztecEditor-Android/pull/461) through the `OnMediaDeletedListener` interface that is being listened to by `AztecEditorFragment` within WPAndroid. This in turn calls the host of such fragment, `EditPostActivity`, which takes care of notifying the `UploadService` of such a thing, which in turn takes care of cancelling the upload as it's not needed anymore.

- this comes with one other situation that can come up from this, and it's the user can go back and `undo` this on Aztec, so you end up having the same media item again, but now the upload for such an item has been stopped (and wiped out from the UploadService altogether). The `EditorFragmentListener` now has a new method, `onUndoMediaCheck()`, where the implementation checks for those media items that are in the new Post content (_undoed_ content) marked with the `UPLOADING` class attribute but are not there in the `UploadService`. These are marked as `FAILED` so they can be retried by the user.

To test:
1. start a new draft
2. insert a media item - preferrably something big so you're able to test an in-progress upload cancelation.
3. while the media item is being uploaded, place the cursor to its right (or after it), and hit backspace to delete it
4. observe the upload is stopped
5. tap on the UNDO button
6. observe the media item appears again, now with the "failed"  overlay
7. Tap on it to start the upload again, as normal

Test this also with variants such as: exiting the editor in the middle of an upload (so the upoading notification is triggered) and coming back in again, or inserting several media items mixed with text, selecting them all and deleting it, check how the Posts list behaves, etc.
Maybe even test we haven't broken anything in the Visual Editor as well (as this should only work with Aztec)

cc @nbradbury
